### PR TITLE
Timeseries: Graph series override stack migration

### LINKED ISSUE #67284: Graph -> Timeseries migration: Override value for stacking doesn't migrate properly
## What happened:

When there is an override for stacking (turned off) in a Graph panel and it gets migrated to Timeseries, then stacking gets turned on for that value.

Screenshot after migration:

2023-04-26-145300_392x394_scrot

Notice that the 'Name of stacking group' text box gets filled in with `false`, that's the boolean value from the Graph panel override.

This example comes from this dashboard: 

https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/eed459199703c969afc318ea55b9361ae48180a7/dashboards/resources/pod.libsonnet#L123-L142

## What you expected to happen:

Stacking stays turned off.

## How to reproduce it (as minimally and precisely as possible):

1. Create Graph panel with stacking turned on
2. Add an override for one query value turning off stacking
3. Migrate to Timeseries, notice that stacking is turned on for that value

## Environment:
- Grafana version: Grafana v10.0.0-55070pre (47aa8c7654)

### PR DESCRIPTION
**Pull request by adela-almasan**
The old graph's series override stack migration seemed to always set the stack mode to normal. This PR is updating the migration to take into the account the actual settings.

https://github.com/grafana/grafana/assets/88068998/7dd03578-963c-4560-b250-d203b5857f74

Fixes #67284 

**Testing steps:**
- Create a Graph panel [or use test panel [json](https://github.com/grafana/grafana/files/11480825/graph.json.tx)]
- From Series override section click "Add series override"
- Play around with Stack options - true/false/value
- Migrate to Timeseries
- Expected result: the overrides should match

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/plugins/panel/timeseries/migrations.ts [hunk 1]
```diff
@@ -711,7 +711,7 @@ function migrateHideFrom(panel: {
   }
 }
 
-function getLegendHideFromOverride(reducer: ReducerID.allIsZero | ReducerID.allIsNull) {
+const getLegendHideFromOverride = (reducer: ReducerID.allIsZero | ReducerID.allIsNull) => {
```
Comment (ryantxu) [inline public/app/plugins/panel/timeseries/migrations.ts:714]: 6 vs 12/dozen... but why use const for simple functions?  There are cases where setting implementation is nice, but in general I vote we keep functions functions when they are just functions

**Review by nmarrs — APPROVED** [review]
tested locally and works well 🥧

File: public/app/plugins/panel/timeseries/migrations.ts (hunk 1 above)
Comment (adela-almasan) [reply public/app/plugins/panel/timeseries/migrations.ts:714]: Hm, personal preference in general, I don't have strong opinions about them, I can switch back if that works better for everybody..

File: public/app/plugins/panel/timeseries/migrations.ts [hunk 2]
```diff
@@ -732,4 +732,11 @@ function getLegendHideFromOverride(reducer: ReducerID.allIsZero | ReducerID.allI
       },
     ],
   };
-}
+};
+
+const getStackingFromOverrides = (value: Boolean | string) => {
+  return {
+    mode: value ? StackingMode.Normal : StackingMode.None,
+    group: isString(value) ? value : 'A',
```
[the remark this answers is not in the document]
Comment (adela-almasan) [reply public/app/plugins/panel/timeseries/migrations.ts]: It's the stacking group name. Pushed an update.

File: public/app/plugins/panel/timeseries/migrations.ts (hunk 1 above)
Comment (nmarrs) [reply public/app/plugins/panel/timeseries/migrations.ts]: This is an interesting [quick take on it](https://dev.to/skinnypetethegiraffe/function-or-const-which-do-you-prefer-typescriptjavascript-apa). I agree slightly on readability of knowing quickly what is a function vs variable but it seems like a very minor thing. I'm ok with letting devs do whatever fits their personal preference 🤷 

maybe though that means leaving already made `function` as functions and visa versa? 😬

File: public/app/plugins/panel/timeseries/migrations.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (nmarrs) [reply public/app/plugins/panel/timeseries/migrations.ts]: LGTM :shipit:

**Pull request merged**